src/main/java/org/opensearch/flowframework/workflow/CreateIngestPipelineStep.java

```diff
@@ -187,12 +112,75 @@ public PlainActionFuture<WorkflowData> execute(
                     logger.error(errorMessage, e);
                     createIngestPipelineFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
                 }
+            }
 
-            }, exception -> {
+            @Override
+            public void onFailure(Exception e) {
                 String errorMessage = "Failed to create ingest pipeline";
-                logger.error(errorMessage, exception);
-                createIngestPipelineFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(exception)));
-            }));
+                logger.error(errorMessage, e);
+                createIngestPipelineFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
+            }
+
+        };
+
+        Set<String> requiredKeys = Set.of(PIPELINE_ID, CONFIGURATIONS);
+
+        // currently, we are supporting an optional param of model ID into the various processors
+        Set<String> optionalKeys = Set.of(MODEL_ID);
+
+        try {
+            Map<String, Object> inputs = ParseUtils.getInputsFromPreviousSteps(
+                requiredKeys,
+                optionalKeys,
+                currentNodeInputs,
+                outputs,
+                previousNodeInputs,
+                params
+            );
+
+            String pipelineId = (String) inputs.get(PIPELINE_ID);
```

Comment:
Perhaps we can fail this earlier by checking if the pipeline ID is null here. If it is, then we shouldn't attempt substitution on the configuration since the request will fail either way.

Reply:
@joshpalis that null check is done in the `getInputsFromPreviousSteps` since that's a required key.

Reply:
removed null checks as they happen earlier